### PR TITLE
Updated set-java-home.bash to only update if the working directory has changed

### PR DESCRIPTION
**Pull request by Johnny-Malizia**
* Updated set-java-home.bash to only update if the working directory has changed

Without this change, there is a very noticeable latency added to the terminal.

### CODE REVIEW

**Comment by jmalizia-branch** [comment]
@halcyon is there anything I need to do to help get this merged?

File: set-java-home.bash [hunk 1]
```diff
@@ -8,6 +8,10 @@ asdf_update_java_home() {
 }
 
 prompt_command() {
+  if [ "$PWD" == "$LAST_PWD" ]; then
```
Comment (halcyon) [inline set-java-home.bash:11]: Please use `[[` and `]]`, also please surround expanded variables with `{}` - e.g. "${PWD}"

File: set-java-home.bash [hunk 2]
```diff
@@ -8,6 +8,10 @@ asdf_update_java_home() {
...
+    return
+  fi
+  LAST_PWD=$PWD
```
Comment (halcyon) [inline set-java-home.bash:14]: Please use "${PWD}"

**Comment by halcyon** [comment]
@Johnny-Malizia much appreciated!

**Comment by jmalizia-branch** [comment]
No problem!

**Comment by halcyon** [comment]
LGTM!

**Comment by halcyon** [comment]
@Johnny-Malizia I just tried it out, and am a little concerned that this change does come with a trade off. Changing JDKs isn't directly related to changing one's working directory. It seems completely plausible to me that I might want to change JDKs and not change the working directory.

**Comment by jmalizia-branch** [comment]
I understand, and it's ok if you want to revert it. I'll look for an alternative solution. I think we should be setting the `JAVA_HOME` when the java binary is invoked not at every shell prompt. That would also fix the problem, but I'm trying to see how to do that.

**Comment by Johnny-Malizia** [comment]
I understand and if you want to revert the change I'm ok with that. I think we can solve this by exporting the `JAVA_HOME` when the java binary is invoked by setting up `exec-path` to return the path, but also set the env var. This would add minimal latency to invoking the java binary, and remove the overhead from the shell_prompt.

**Pull request merged**